9_report/src/fig_input_data.R

```diff
@@ -18,27 +18,65 @@ fig_input_data <- function(fig_ind, input_example_yml, preds_ind, remake_file, c
               nwm_forecast_ind = '2_munge/out/agg_nwm_long1.rds.ind',
               remake_file = '3_forecast.yml')
 
+  flux_nwis <- eList$Sample %>%
+    dplyr::filter(Uncen == 1) %>% # only two data points that are censored (removing for now)
+    select(Date, Q, ConcAve) %>%
+    mutate(Flux = Q * ConcAve * 60*60*24/1000, data = 'nwis') %>% # flux in kg N day-1
+    rename(Conc = ConcAve)
+
+  preds <- preds %>%
+    select(Date, Flow, Conc, Flux) %>%
+    rename(Q = Flow) %>%
+    mutate(data = 'forecast') %>%
+    bind_rows(flux_nwis, .)
+
   xlim = range(c(eList$Daily$Date, eList$Sample$Date, preds$Date))
 
   # flow
-  g1 <- ggplotGrob(ggplot(data = eList$Daily, aes(x = Date, y = Q)) +
-    geom_line() + theme_classic() + xlim(xlim) +
-      theme(axis.title.x = element_blank()))
+  g1 <- ggplotGrob(ggplot(data = preds, aes(x = Date, y = Q)) +
+                     geom_line(size = 1,
+                               aes(color = factor(data))) +
+                     scale_color_manual(name = 'data',
+                                        values = c('nwis' = 'black',
+                                                   'forecast' = 'red'),
+                                        labels = c('Forecast', 'Retro')) +
+                     theme_classic() +
+                     xlim(xlim) +
+                     labs(y = expression(Discharge~(m^3~s^-1))) +
+                     theme(axis.title.x = element_blank(),
+                           legend.position = c(.1,.9), legend.title = element_blank()) +
+                     geom_vline(xintercept = as.Date(input_ex$ref_date), linetype = 'dashed'))
 
   #concentration
-  g2 <- ggplotGrob(ggplot(data = eList$Sample, aes(x = Date, y = ConcAve)) +
-    geom_point() + theme_classic()+ xlim(xlim) +
-      theme(axis.title.x = element_blank()))
+  g2 <- ggplotGrob(ggplot(data = preds, aes(x = Date, y = Conc)) +
+                     geom_point(size = 1,
+                                aes(colour = factor(data))) +
+                     scale_color_manual(name = 'data',
+                                        values = c('nwis' = 'black',
+                                                   'forecast' = 'red')) +
+                     theme_classic()+
+                     xlim(xlim) +
+                     labs(y = expression(Nitrate~concentration~(mg~N~L^-1))) +
+                     theme(axis.title.x = element_blank(), legend.position = 'none') +
+                     geom_vline(xintercept = as.Date(input_ex$ref_date), linetype = 'dashed'))
 
   #loads
   g3 <- ggplotGrob(ggplot(data = preds, aes(x = Date, y = Flux)) +
                      xlim(xlim) +
-                     geom_line() + theme_classic())
+                     geom_line(size = 1,
+                               aes(colour = factor(data))) +
```

Comment:
Some of the days in the calibration period lack concentration data but look from this line as if they have flux data. One way to clarify would be to plot fluxes as points (at least up to the forecast period, and then I'm open to either points or lines) so that the gaps are as visible as they are for concentrations.